### PR TITLE
Suppress belongsto index

### PR DESCRIPTION
**Pull request by poojamat**
*Issue #, if available:*

*Description of changes:*
Suppress creation of index based on @index annotation when the index fields occur in belongs to association as there is an index created on foreign key already.
*How did you test these changes?*
(Please add a line here how the changes were tested)

- [X ] Added Unit Tests
- [ ] Tested on frontend (Add Screenshots)
- [ ] Successful logs (Attach them to the PR)

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java [hunk 1]
```diff
@@ -136,15 +136,20 @@ private <T extends Model> Completable processOutboxItem(PendingMutation<T> mutat
...
-                    // Once the server knows about it, it's safe to remove from the outbox.
-                    // This is done before merging, because the merger will refuse to merge
-                    // if there are outstanding mutations in the outbox.
-                    mutationOutbox.remove(mutationOutboxItem.getMutationId())
-                        .andThen(merger.merge(modelWithMetadata))
-                        .doOnComplete(() -> {
-                            String modelName = mutationOutboxItem.getModelSchema().getName();
-                            announceMutationProcessed(modelName, modelWithMetadata);
-                        })
+                            // Once the server knows about it, it's safe to remove from the outbox.
```
Comment (mikepschneider) [inline aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java:139]: nit: looks like the indentation is off here

File: aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLCommandProcessorTest.java [hunk 2]
```diff
@@ -166,4 +167,15 @@ public void executeExistsReturnsFalseWhenItemDoesntExist() throws AmplifyExcepti
         SqlCommand existsCommand = sqlCommandFactory.existsFor(blogOwnerSchema, predicate);
         assertFalse(sqlCommandProcessor.executeExists(existsCommand));
     }
+
+    /**
+     * Create a BlogOwner, but don't insert it.  Then verify that executeExists returns false.
+     * @throws AmplifyException on failure to create ModelSchema from class.
+     */
+    @Test
+    public void testIndexNotCreatedWhenFieldsInBelongsTo() throws AmplifyException {
```
Comment (mikepschneider) [inline aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLCommandProcessorTest.java:176]: Do you have adequate testing for this?  I only see that it’s testing that one index is created, but it doesn’t check the contents of that index.

Are there any other edge cases that should be checked?

Also the comment says `Then verify that executeExists returns false` but I don’t see that in the test.

Comment (poojamat) [reply aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLCommandProcessorTest.java:177]: Resolved.

File: aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java (hunk 1 above)
Comment (poojamat) [reply aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java:139]: Resolved

**Review by mikepschneider — APPROVED** [review]
